pkg/apis/vendors/v1alpha1/taxid_utils.go

```diff
@@ -0,0 +1,163 @@
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetTaxIdFromSecret retrieves the tax ID from the referenced Secret
+func GetTaxIdFromSecret(ctx context.Context, c client.Client, vendor *Vendor, taxIdRef TaxIdReference) (string, error) {
+	// Determine the namespace for the Secret
+	namespace := taxIdRef.Namespace
+	if namespace == "" {
+		namespace = vendor.Namespace
+		if namespace == "" {
+			namespace = "default" // fallback for cluster-scoped resources
+		}
+	}
+
+	// Get the Secret
+	secret := &corev1.Secret{}
+	secretKey := types.NamespacedName{
+		Name:      taxIdRef.SecretName,
+		Namespace: namespace,
+	}
+
+	if err := c.Get(ctx, secretKey, secret); err != nil {
+		return "", fmt.Errorf("failed to get secret %s/%s: %w", namespace, taxIdRef.SecretName, err)
+	}
+
+	// Extract the tax ID from the Secret
+	taxIdBytes, exists := secret.Data[taxIdRef.SecretKey]
+	if !exists {
+		return "", fmt.Errorf("key %s not found in secret %s/%s", taxIdRef.SecretKey, namespace, taxIdRef.SecretName)
+	}
+
+	return string(taxIdBytes), nil
+}
```

Comment:
We should leave the `pkg/apis/vendors` package to API type definitions only. Any helper functions like this should be put in the controller package where they're used.